tdp/cli/commands/playbooks.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2022 TOSIT.IO
+# SPDX-License-Identifier: Apache-2.0
+
+from pathlib import Path
+
+import click
+import networkx as nx
+
+from tdp.cli.context import pass_dag
+from tdp.core.component import Component
+from tdp.core.dag import DEFAULT_SERVICE_PRIORITY, SERVICE_PRIORITY
+
+
+@click.command(
+    short_help="Generate meta playbooks in order to use tdp-collection without tdp-lib"
+)
+@click.argument("services", required=False, nargs=-1)
+@click.option(
+    "--output-dir",
+    type=Path,
+    help="Output dir where playbooks will be written",
+    required=False,
+    default=".",
+)
+@pass_dag
+def playbooks(dag, services, output_dir):
+    # services DAG
+    dag_services = nx.DiGraph()
+    # For each service, get all components with DAG topological_sort order
+    dag_services_components = {}
+    for component_name in dag.get_all_actions():
+        component = dag.components[component_name]
+        dag_services.add_node(component.service)
+        for dependency in component.depends_on:
+            dependency_component = Component(dependency)
+            if dependency_component.service != component.service:
+                dag_services.add_edge(dependency_component.service, component.service)
+        dag_services_components.setdefault(component.service, []).append(component)
+
+    if not nx.is_directed_acyclic_graph(dag_services):
+        raise RuntimeError("dag_services is not a DAG")
+
+    def custom_key(node):
+        component_priority = SERVICE_PRIORITY.get(node, DEFAULT_SERVICE_PRIORITY)
+        return f"{component_priority:02d}_{node}"
+
+    dag_services_order = nx.lexicographical_topological_sort(dag_services, custom_key)
```

Comment:
@gboutry I have a problem with the order.
What I get:
```
['zookeeper', 'hadoop', 'ranger', 'knox', 'exporter', 'hdfs', 'yarn', 'hive', 'hbase', 'spark']
```

What we want :
```
['zookeeper', 'hadoop', 'ranger', 'exporter', 'hdfs', 'yarn', 'hive', 'hbase', 'spark', 'knox']
```
For testing I use `nx.all_topological_sorts(dag_services)` which give me 150 topological ordering, and the one I want is one of them.

The problem may occur because we do not specify a priority for `exporter` service which default to 99.

Reply:
If I add `"exporter": 0,` inside `SERVICE_PRIORITY` I get:
```
['exporter', 'zookeeper', 'hadoop', 'ranger', 'hdfs', 'yarn', 'hive', 'hbase', 'spark', 'knox']
```
Maybe it is the solution.

Reply:
Settings exporter to 0 is an acceptable quickfix for now. But what happens if we add another auxiliary service ? Is it acceptable to put every auxiliary services to priority `0` ? Maybe it is

Reply:
Considering priority 0 as "before everything else" I think it's acceptable

Reply:
#141 will fix it.